Improve AttachmentFile implementations & add WikiService.getAllContent

labkey-alan opened the pull request:
#### Rationale
This PR does a few things that are needed for the improved embedded attachment behavior in ELN (see changes below).

#### Related Pull Requests
* https://github.com/LabKey/biologics/pull/1239
* https://github.com/LabKey/labbook/pull/155

#### Changes
* Make the behavior for openInputStream and closeInputStream consistent across all AttachmentFile implementations
  * InputStreamAttachmentFile is the one exception, because it cannot be re-opened.
* Adds getAllContent to WikiService so modules can fetch the content for every version of a given wiki
  * This is needed so we can search previous versions for attachments and remove any attachments that do not exist in any version of a wiki.

labkey-alan committed:
ByteArrayAttachmentFile: reset _inputStream during closeInputStream()

labkey-alan committed:
Make behavior consistent across AttachmentFile implementations

labkey-alan committed:
WikiService: add getAllContent

labkey-nicka commented on api/src/org/labkey/api/wiki/WikiService.java:
```diff
@@ -69,6 +69,11 @@ static void setInstance(WikiService impl)
+    /**
+     * Returns the raw (unformatted) content for every version of the wiki
+     */
+    List<String> getAllContent(Container c, String wikiName);
```
Mark this interface as `@Nullable` since it could return `null` in the case the wiki itself is not found.

labkey-alan committed:
WikiService: Mark getContent/getAllContent Nullable

labkey-alan committed:
AbstractQueryImportAction: Catch exception when trying to close file input stream

Sometimes the input stream was never opened, so we can ignore the exception when trying to close it

labkey-alan committed:
AttachmentDataIterator Fix exception handling in next() method
